app/javascript/components/common/Editor/TestRuns.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react'
+import { Submission, SubmissionTestsStatus } from '../Editor'
+import consumer from '../../../utils/action-cable-consumer'
+
+function Content({ submission }: { submission: Submission }) {
+  switch (submission.testsStatus) {
+    case SubmissionTestsStatus.PASS:
+    case SubmissionTestsStatus.FAIL:
+      return (
+        <>
+          {submission.testRuns.map((run) => (
+            <p key={run.name}>
+              name: {run.name}, status: {run.status}, output: {run.output}
+            </p>
+          ))}
+        </>
+      )
+    case SubmissionTestsStatus.ERROR:
+    case SubmissionTestsStatus.OPS_ERROR:
+      return <p>{submission.message}</p>
+    default:
+      return <></>
+  }
+}
+
+export function TestRuns(props: { submission: Submission }) {
+  const [submission, setSubmission] = useState(props.submission)
+
+  useEffect(() => {
+    const subscription = consumer.subscriptions.create(
+      { channel: 'Test::Submissions::TestRunsChannel', id: submission.id },
+      {
+        received: (json: any) => {
+          setSubmission({
+            id: submission.id,
+            testsStatus: json.tests_status,
+            testRuns: json.test_runs,
+            message: json.message,
+          })
```

Comment:
@kntsoriano So I think the bit you've misunderstood here (misunderstood from what I meant when explaining) is that this won't be a submission coming back, it'll be a serialized test run.

This is why I'm confused about the `message` bit in Submission as you've sort of merged two serailizations into one.

Reply:
Just one test run? I based the JSON from here: https://github.com/exercism/v3-project-management/issues/52#issue-715234576